js/apps/system/_admin/aardvark/APP/react/src/views/query/editor/QueryOptionsTab.tsx

```diff
@@ -0,0 +1,317 @@
+              })}
+            </Grid>
+          </AccordionPanel>
+        </AccordionItem>
+        <AccordionItem borderColor="gray.300">
+          <AccordionButton>
+            <Box as="span" flex="1" textAlign="left" fontWeight="bold">
+              Optimizer Rules
+            </Box>
+            <AccordionIcon />
+          </AccordionButton>
+
+          <AccordionPanel pb={4}>
+            <OptimizerRules />
+          </AccordionPanel>
+        </AccordionItem>
+      </Accordion>
+    </Box>
+  );
+};
+
+const QueryOptionRow = ({
+  queryOptionKey,
+  options
+}: {
+  queryOptionKey: string;
+  options: {
+    [key: string]: any;
+  };
+}) => {
+  const { queryOptions, setQueryOptions } = useQueryContext();
+  const value = queryOptions[queryOptionKey];
+  const option = options[queryOptionKey];
+  if (option.type === "boolean") {
+    return (
+      <>
+        <FormLabel>{option.label}</FormLabel>
+        <Switch
+          size="sm"
+          isChecked={value}
+          onChange={e => {
+            const updatedValue = e.target.checked;
+            setQueryOptions(prev => {
+              return { ...prev, [queryOptionKey]: updatedValue };
+            });
+          }}
+        />
+      </>
+    );
+  }
+  return (
+    <>
+      <FormLabel>{option.label}</FormLabel>
+      <Input
+        size="xs"
+        value={value}
+        // this is required to override bootstrap styles
+        css={{
+          height: "24px !important"
+        }}
+        type={option.type}
+        onChange={e => {
+          const updatedValue = e.target.value;
+          setQueryOptions(prev => {
+            return { ...prev, [queryOptionKey]: updatedValue };
+          });
+        }}
+      />
+    </>
+  );
+};
+type OptimizerRule = {
+  name: string;
+  flags: {
+    canBeDisabled: boolean;
+    canCreateAdditionalPlans: boolean;
+    clusterOnly: boolean;
+    disabledByDefault: boolean;
+    enterpriseOnly: boolean;
+    hidden: boolean;
+  };
+};
+const useFetchQueryOptimizerRuleOptions = () => {
+  const [optimizerRules, setOptimizerRuleOptions] = React.useState<
+    OptionType[]
+  >([]);
+  const isEnterprise = window.frontendConfig.isEnterprise;
+  const isCluster = window.frontendConfig.isCluster;
+  React.useEffect(() => {
+    const fetchRules = async () => {
+      const currentDB = getCurrentDB();
+      const response = await currentDB.request(
+        {
+          method: "GET",
+          path: "/_api/query/rules"
+        },
+        res => res.body
+      );
+      const ruleOptions = response
+        .filter((rule: OptimizerRule) => {
+          if (rule.flags.enterpriseOnly && !isEnterprise) {
+            return false;
+          }
+          if (rule.flags.clusterOnly && !isCluster) {
+            return false;
+          }
+          if (!rule.flags.canBeDisabled) {
+            return false;
+          }
+          if (rule.flags.hidden) {
+            return false;
+          }
+          return true;
```

Comment:
It could be done in a single sentence:

`!(rule.flags.enterpriseOnly && !isEnterprise || rule.flags.clusterOnly && !isCluster || !rule.flags.canBeDisabled || rule.flags.hidden)`

The only inconvenient is that loses some readability. Just as a suggestion, up to you to change it.

Reply:
I'd suggest always choosing readability in cases like these, instead of doing it in a single line. Would prefer to keep it this way, and I have further made it more readable by pulling out the flags early, so we don't have to keep doing `rules.flags.{flag}`